pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -270,6 +271,33 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			}
 		}
 	}
+
+	// Deploying test based on the storageclassDeviceSet
+	// Todo: Replace the below logic to deploy osd pods over pvcs
+	for _, storageclassDeviceSet := range c.DesiredStorage.Storage {
+		for i := 0; i < storageclassDeviceSet.Count; i++ {
+			sc := "test-sc"
+			pvc := &v1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pvc-" + storageclassDeviceSet.Name + fmt.Sprintf("-%v", i),
+					Namespace: "rook-ceph",
+				},
+				Spec: v1.PersistentVolumeClaimSpec{
```

Comment:
Here you don't want to just statically create the struct. You want to take `sds := storageClassDeviceSets[i].volumeClaimTemplates[0]` as it's own var, modify any fields that are not already specified with our defaults (e.g. access modes, volume mode block), and then set `Spec: sds,`.

Reply:
yup will change it, its wrong